### PR TITLE
feat(backend/frontend/shared): Add pagination to the Activities page tm-646

### PR DESCRIPTION
**Pull request by GEOFARL**

### CODE REVIEW

File: apps/backend/src/modules/activities/activity.repository.ts [hunk 1]
```diff
@@ -242,6 +247,78 @@ class ActivityRepository implements Repository<ActivityEntity> {
 		});
 	}
 
+	public async findAllWithPagination({
```
Comment (what1s1ove) [inline apps/backend/src/modules/activities/activity.repository.ts:250]:
```suggestion
	public async findAllWith({
```
should be with params. it's by default with pagination

File: apps/frontend/src/pages/feed/libs/components/feed-activity-list/feed-activity-list.tsx [hunk 2]
```diff
@@ -11,20 +12,61 @@ import styles from "./styles.module.css";
...
+		[handleIncreasePage, isLoadingMore],
+	);
+
 	return (
 		<div className={styles["feed"]}>
 			{activities.map((activity) => (
 				<FeedActivity activity={activity} key={activity.id} userId={user.id} />
 			))}
+
+			{activities.length !== totalCount && (
```
Comment (what1s1ove) [inline apps/frontend/src/pages/feed/libs/components/feed-activity-list/feed-activity-list.tsx:65]: add constant

File: apps/frontend/src/pages/feed/libs/components/feed-activity-list/feed-activity-list.tsx [hunk 3]
```diff
@@ -11,20 +12,61 @@ import styles from "./styles.module.css";
...
+}: Properties) => {
 	const { user } = useAppSelector(({ auth }) => {
 		return {
 			user: auth.user as UserAuthResponseDto,
 		};
 	});
 
+	const observer = useRef<IntersectionObserver | null>(null);
+
+	const loaderElementReference = useCallback(
```
Comment (what1s1ove) [inline apps/frontend/src/pages/feed/libs/components/feed-activity-list/feed-activity-list.tsx:34]: add a custom hook for it

File: apps/frontend/src/modules/activities/activities-api.ts [hunk 4]
```diff
@@ -22,17 +23,27 @@ class ActivitiesApi extends BaseHTTPApi {
 		super({ baseUrl, http, path: APIPath.ACTIVITIES, storage });
 	}
 
-	public async getActivities(): Promise<ActivityGetAllResponseDto> {
+	public async getActivities(
```
Comment (what1s1ove) [inline apps/frontend/src/modules/activities/activities-api.ts:26]:
```suggestion
	public async getAll(
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/backend/src/modules/activities/activity.repository.ts [hunk 5]
```diff
@@ -242,6 +247,78 @@ class ActivityRepository implements Repository<ActivityEntity> {
 		});
 	}
 
+	public async findAllWithParams({
```
Comment (what1s1ove) [inline apps/backend/src/modules/activities/activity.repository.ts:250]:
```suggestion
	public async findAll({
```

File: apps/frontend/src/libs/hooks/use-infinite-scroll/use-infinite-scroll.hook.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import { useCallback, useRef } from "react";
+
+type Properties = {
+	isLoading: boolean;
+	loadMore: () => void;
```
Comment (what1s1ove) [inline apps/frontend/src/libs/hooks/use-infinite-scroll/use-infinite-scroll.hook.ts:5]:
```suggestion
	onLoadMore: () => void;
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/backend/src/modules/activities/activity.repository.ts (hunk 5 above)
Comment (GEOFARL) [reply apps/backend/src/modules/activities/activity.repository.ts:250]: image
image
image

Comment (what1s1ove) [reply apps/backend/src/modules/activities/activity.repository.ts:250]: implement them. it should be this route, right

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/backend/src/modules/activities/activity.repository.ts (hunk 5 above)
Comment (GEOFARL) [reply apps/backend/src/modules/activities/activity.repository.ts:250]: @what1s1ove, but how should I implement them? FindAll method expects to return only the array of entities, not the response in the pagination format with items and total. No repository that supports pagination implements it in the findAll method

Comment (GEOFARL) [reply apps/backend/src/modules/activities/activity.repository.ts:250]: @what1s1ove, but what to do with `total` in that case? I need it either

Comment (what1s1ove) [reply apps/backend/src/modules/activities/activity.repository.ts:250]: you can add it to the return type in the repository. what is the problem?

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**